skyvern/webeye/browser_factory.py

```diff
@@ -122,8 +128,27 @@ async def _create_headful_chromium(playwright: Playwright, **kwargs: dict) -> tu
     return browser_context, browser_artifacts
 
 
+async def _create_attached_chromium(playwright: Playwright, ws_endpoint: str, **kwargs: dict) -> tuple[BrowserContext, BrowserArtifacts]:
+    browser = await playwright.chromium.connect_over_cdp(ws_endpoint)
+    browser_args = BrowserContextFactory.build_browser_args()
+    browser_args.update(
+        {
+            "headless": False,
+        }
+    )
+    contexts = browser.contexts
+    if contexts:
+        browser_context = contexts[0]
+    else:
+        print("No existing contexts found. Creating a new context...")
+        browser_context = await browser.new_context(**browser_args)
+
+    browser_artifacts = BrowserContextFactory.build_browser_artifacts(har_path=browser_args["record_har_path"])
+    return browser_context, browser_artifacts
```

Comment:
What's the error you get trying this out??

Reply:
This code generally looks correct to me (minus the `if contexts` part)

Reply:
Resharing here:

```2024-03-21T03:26:11.616961Z [info     ] Creating browser state for task task_id=tsk_237547645102207568
2024-03-21T03:26:13.376947Z [info     ] Creating a new page
2024-03-21T03:26:13.917403Z [info     ] A new page is created
2024-03-21T03:26:13.917509Z [info     ] Navigating page to https://www.finditparts.com and waiting for 3 seconds
2024-03-21T03:26:17.559109Z [info     ] Successfully went to https://www.finditparts.com```

Essentially it opens the page and then just stops there.

Is there any log setting to modify to get more verbose logging?

Reply:
Are you connecting over CDP locally? If you turn headless=true does a new window actually open up?

Reply:
Debugging playwright has definitely been a nightmare for us

Reply:
Yes, connecting locally.

1. Kill all running chromium processes
2. Launch a new window by adding the `--remote-debugging-port=9222`

That pointed me in the right direction. It looks like a lot of the settings needed (like har path) can't be set on an existing browser_context: https://playwright.dev/python/docs/api/class-browsercontext

Although, if I remove the check for using an existing browser context it doesn't launch at all.